src/main/groovy/org/asciidoctor/gradle/AsciidoctorTask.groovy

```diff
@@ -451,11 +453,11 @@ class AsciidoctorTask extends DefaultTask {
      * @since 1.5.1
      */
     @OutputDirectories
-    Set<File> getOutputDirectories() {
+    Map<String, File> getOutputDirectories() {
```

Comment:
This will probably break some build scripts. `@OutputDirectories` can annotate a method that returns Map<String,File>` or `Iterable<File>`. `Set<File>` is an implementation of the latter, so I am not sure what the chaneg is really for.

Reply:
This would be necessary because only @OutputDirectories that return a Map can be cached. See https://docs.gradle.org/4.1/userguide/more_about_tasks.html#sec:task_inputs_outputs

However, if you do not want to break this, one could for instance add a separate _cacheable_ subclass . See my general comment below.

Reply:
How about we leave that method as is, and annotate it with internal. Then we can add a separate method returning the `Map<String, File>` which is annotated with `@OutputDirectories`. WDYT?